YatimaStdLib/List.lean

```diff
@@ -39,4 +39,83 @@ instance : Foldable List where
   foldr := List.foldr
   foldl := List.foldl
 
+-- the `B` suffix avoids name conflicts with mathlib
+def eraseDupB [BEq α] : List α → List α
+  | [] => []
+  | x::xs => 
+    let exs := eraseDupB xs
+    if exs.contains x then exs else x::exs
+
+def splitAtP [BEq α] (p : α → Bool) (l : List α) : List α × List α :=
+  match l.dropWhile p with 
+  | [] => unreachable!
+  | a::as => ⟨l.takeWhile p ++ [a], as⟩
+
+partial def mergeM [Monad μ] (cmp: α → α → μ Ordering) : List α → List α → μ (List α)
+  | as@(a::as'), bs@(b::bs') => do
+    if (← cmp a b) == Ordering.gt
```

Comment:
From the signature alone, it's not clear which elements take precedence -- those greater or those lesser. I've heard rumours from @arthurpaulino that default arguments exist in Lean. Perhaps your mergeM can take `leftPrecedes` argument, defaulting to `Ordering.gt`? Now people won't need to read the implementation to understand what's up.